index.html

```diff
@@ -11861,9 +11861,14 @@ <h2>Definitions of States and Properties (all aria-* attributes)</h2>
-				<p>Authors MAY, with caution, use aria-hidden to hide visibly rendered content from assistive technologies <em>only</em> if the act of hiding this content is intended to improve the experience for users of assistive technologies by removing redundant or extraneous content. Authors using aria-hidden to hide visible content from screen readers MUST ensure that identical or equivalent meaning and functionality is exposed to assistive technologies.</p>
+				<p>While authors do not need to use <code>aria-hidden</code> when visibly hiding content with CSS, </code>authors MAY, with caution, use <code>aria-hidden="true"</code> to hide visibly rendered content from assistive technologies. However, authors should <em>only</em> do so if the act of hiding this content is intended to improve the experience for users of assistive technologies (e.g., to remove redundant or extraneous content). Authors using <code>aria-hidden="true"</code> to hide visible content from screen readers MUST ensure that identical or equivalent meaning and functionality is exposed to assistive technologies.</p>
+				<p>User agents MUST either prune <code>aria-hidden</code> nodes from the accessibility tree, or mark all <code>aria-hidden</code> nodes as hidden.</p>
+				<p>Authors MAY use <code>aria-hidden="false"</code> within an <code>aria-hidden="true"</code> subtree to unhide the visibly rendered content within the <code>aria-hidden="false"</code> subtree.</p>
+				<p class="note">This is a change from ARIA 1.2 and earlier versions in several ways: 1) <code>aria-hidden="false"</code> no longer un-hides content that is also hidden in CSS, as there are other techniques to present content only to AT users that works more consistently (e.g. positioning offscreen), and 2) <code>aria-hidden="false"</code> previously was a <code>noop</code> on (e.g., did not affect) visible content, and 3) <code>aria-hidden="false"</code> previously operated only on a single element. However, the <code>aria-hidden="false"</code> behavior of previous versions was never consistently implemented, understood or commonly used.</p>
+				<p>Authors MUST NOT place focusable elements within <code>aria-hidden</code> content. This is because the AT will have no information about the content, so it cannot be presented to the user with AT. User agents MUST repair this situation by ignoring the <code>aria-hidden</code> attribute on the ancestor of the focused element from that point forward. When this repair is made, user agents SHOULD inform authors through developer tools (if available).</p>
```

Comment:
I think the wording got messed up on this one. It's intended to say that Authors MUST not allow elements within <code>aria-hidden</code> content to obtain focus, because there is more than one way of doing that, right? And it makes it clear that scrollable focusable areas are included.

For #2, it is meant to apply to elements that actually ended up receiving focus. At that point forward, aria-hidden="true" will be ignored on the entire subtree that has it from that point forward, even if the element within it loses focus. I agree it's not going to solve everything, but that's why we have multiple mitigations. There are definitely scenarios where something gets focus when the user has a screen reader and is navigating with a virtual buffer (e.g. the author places it there because the user clicked an open dialog button).

Reply:
@MelSumner in our initial draft, I have different text. I have:

Authors MUST prevent elements from receiving focus within aria-hidden content, because when AT users reach this content, the AT can provide the user with no information about the content, leading to a disorienting and inaccessible experience. User agents MUST repair this situation by ignoring aria-hidden on the ancestor of the focused element.

I think it's better because it may be very difficult to avoid having focusable elements. But if the user can't actually focus the element via input means, maybe that's good enough. E.g. it's not tabbable, not clickable and hidden from AT, that prevents it from being focused.

Reply:
Thinking about it more, I guess it depends what is meant by focusable. I would thing that an <input> will be considered focusable, but if I have code that prevents it from being focused, or redirects focus if it does occur — either seems fine, right? I don't think we want to ask authors to have to go through their aria-hidden content and apply tabindex=-1. Interested in other opinions on this wording.

Reply:
Ah, I misunderstood the intent here. The repair is only intended to apply to focused elements, so the text is correct there. However, the mention of focusable elements earlier (which is effectively author requirement, not UA repair) confused me.

Perhaps we could say:

`If an element inside an aria-hidden ancestor receives focus, user agents MUST repair this situation by ignoring the <code>aria-hidden</code> attribute on the ancestor of the focused element from that point forward.`

I didn't think of the "open dialog" style example. Thanks for that. As a side note, this is one reason I'd really like to see solid use cases being included alongside ARIA spec changes (probably not in the spec itself, but at least in the PR). This makes it a lot easier as a reviewer to understand why a change is being made and would avoid a lot of misunderstandings like this. It would also make it a lot easier when looking at historical changes to figure out why the change was made.